plugins/meta/route-override/route-override.go

```diff
@@ -0,0 +1,419 @@
+	return netlink.RouteAdd(&netlink.Route{
+		LinkIndex: dev.Attrs().Index,
+		Scope:     netlink.SCOPE_UNIVERSE,
+		Dst:       &route.Dst,
+		Gw:        route.GW,
+	})
+}
+
+func processRoutes(netnsname string, conf *RouteOverrideConfig) (*current.Result, error) {
+	netns, err := ns.GetNS(netnsname)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open netns %q: %v", netns, err)
+	}
+	defer netns.Close()
+
+	res, err := current.NewResultFromResult(conf.PrevResult)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert result to current version: %v", err)
+	}
+
+	if conf.FlushGateway {
+		// add "0.0.0.0/0" into delRoute to remove it from routing table/result
+		_, gwRoute, _ := net.ParseCIDR("0.0.0.0/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+		_, gwRoute, _ = net.ParseCIDR("::/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+
+		// delete given gateway address
+		for _, ips := range res.IPs {
+			if ips.Version == "6" {
+				ips.Gateway = net.IPv6zero
+			} else {
+				ips.Gateway = net.IPv4zero
+			}
+		}
+	}
+
+	newRoutes := []*types.Route{}
+	err = netns.Do(func(_ ns.NetNS) error {
+		// Flush routes if required
+		if !conf.FlushRoutes {
+		NEXT:
+			for _, route := range res.Routes {
+				for _, delroute := range conf.DelRoutes {
+					if route.Dst.IP.Equal(delroute.Dst.IP) &&
+						bytes.Equal(route.Dst.Mask, delroute.Dst.Mask) {
+						err = deleteRoute(delroute, res)
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "failed to delte route %v: %v", delroute, err)
+						}
+						continue NEXT
+					}
+
+				}
+				newRoutes = append(newRoutes, route)
+			}
+		} else {
+			err = deleteAllRoutes(res)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to delete all routes: %v", err)
+			}
+		}
+
+		if conf.FlushGateway {
+			err = findAndDeleteDefaultRoute(res)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to flush gateway: %v", err)
+			}
+		}
+
+		// Get container IF name
+		var containerIFName string
+		for _, i := range res.Interfaces {
+			if i.Sandbox != "" {
+				containerIFName = i.Name
+				break
+			}
+		}
+		// Add route
+		dev, _ := netlink.LinkByName(containerIFName)
+		for _, route := range conf.AddRoutes {
+			newRoutes = append(newRoutes, route)
+			if err := addRoute(dev, route); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to add route: %v: %v", route, err)
+			}
+		}
+
+		return nil
+	})
+	res.Routes = newRoutes
+
+	return res, nil
+}
+
+func cmdAdd(args *skel.CmdArgs) error {
+	overrideConf, err := parseConf(args.StdinData, args.Args)
+	if err != nil {
+		return err
+	}
+
+	newResult, err := processRoutes(args.Netns, overrideConf)
+	if err != nil {
+		return fmt.Errorf("failed to override routes: %v", err)
+	}
+
+	return types.PrintResult(newResult, overrideConf.CNIVersion)
+}
+
+func cmdDel(args *skel.CmdArgs) error {
+	// TODO: the settings are not reverted to the previous values. Reverting the
```

Comment:
@s1061123 I would say just remove the TODO since I don't think it's useful or feasible to clear out the settings.

Reply:
changed.